core/domain/exp_services_test.py

```diff
@@ -514,6 +514,18 @@ def test_save_and_retrieve_exploration_summary(self):
         self.assertEqual(retrieved_exp_summary.category, 'A new category')
         self.assertEqual(retrieved_exp_summary.contributor_ids, [self.owner_id])
 
+    def test_update_exploration_by_migration_bot(self):
+        self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id, end_state_name='end')
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID)
+
+        exp_services.update_exploration(
```

Comment:
Just to check, does this test fail if the clause is not added in exp_services.py?